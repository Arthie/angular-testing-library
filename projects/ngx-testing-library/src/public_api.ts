/*
 * Public API Surface of ngx-testing-library
 */

export * from './lib/models';
export * from './lib/ngx-testing-library';
