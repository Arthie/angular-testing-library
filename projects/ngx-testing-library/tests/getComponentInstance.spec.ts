import { Component, Input } from '@angular/core';
import { createComponent, fireEvent } from '../src/public_api';

@Component({
  selector: 'fixture',
  template: `<p>rawr</p>  `,
})
class FixtureComponent {}

describe('template syntax', () => {
  test('gets the component instance with a selector', async () => {
    const { getComponentInstance } = await createComponent('<fixture></fixture>', {
      declarations: [FixtureComponent],
    });

    expect(getComponentInstance<FixtureComponent>('fixture')).toBeDefined();
  });

  test('throws an error when no selector is passed', async () => {
    const { getComponentInstance } = await createComponent('<fixture></fixture>', {
      declarations: [FixtureComponent],
    });

    expect(() => getComponentInstance()).toThrowErrorMatchingSnapshot();
  });
});

describe('component type syntax', () => {
  test('gets the component instance', async () => {
    const { getComponentInstance } = await createComponent(
      { component: FixtureComponent },
      {
        declarations: [FixtureComponent],
      },
    );

    expect(getComponentInstance()).toBeDefined();
  });
});
