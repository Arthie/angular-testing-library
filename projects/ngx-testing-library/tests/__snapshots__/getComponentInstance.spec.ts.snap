// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template syntax throws an error when no selector is passed 1`] = `"When using the template syntax, you must provide a selector"`;
